spec/models/generic_file_spec.rb

```diff
@@ -625,4 +625,156 @@
       end
     end
   end
+
+  let(:public_permission_double) { double("Hydra::AccessControls::Permission") }
+  let(:gv_black_photo_sub_collection_double) { double("Collection") }
+  let(:gv_black_collection_double) { double("Collection") }
+  let(:random_doi) { "10.18131/g3-nqgv-4863" }
+  let(:blank_doi) { "" }
+  let(:public_record) { GenericFile.new(doi: []) }
+  let(:private_record) { GenericFile.new(doi: []) }
+
+  # if record is publc, has a doi, and is either in the gv black photograph sub collection or is not in the greater gv black collection
+  describe "#unexportable?" do
+    before do
+      allow(public_permission_double).to receive(:agent).and_return(GenericFile::PUBLIC_PERMISSION)
+
+      allow(gv_black_photo_sub_collection_double).to receive(:map).and_return([GenericFile::GV_BLACK_PHOTOGRAPH_SUB_COLLECTION_ID])
+      allow(gv_black_photo_sub_collection_double).to receive(:id)
+
+      allow(gv_black_collection_double).to receive(:map).and_return([GenericFile::GV_BLACK_COLLECTION_ID])
+      allow(gv_black_collection_double).to receive(:id)
+    end
+
+    context "record is public" do
+      before do
+        # GenericFile validates the type of permissions so simply passing a double to the constructor does not work
+        public_record.stub(:permissions).and_return([public_permission_double])
+      end
+
+      context "has doi" do
+        before do
+          public_record.stub(:doi).and_return(random_doi)
+        end
+
+        context "is in the gv black photograph sub collection" do
+          context "is NOT in the gv black papers collection" do
+            before do
+              public_record.stub(:collections).and_return([gv_black_photo_sub_collection_double])
```

Comment:
Should it be 
`public_record.stub(:collections).and_return(file_collections)`?

Reply:
Yeah, this could probably be cleaned up. I was trying to make sure I covered the different test cases without creating a ton of unnecessary objects, but I think in that process I ended up adding some redundancy.